presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/util/RetryDriver.java

```diff
@@ -125,15 +191,23 @@ public RetryDriver withClassifier(Function<Exception, Exception> classifier)
 
         while (true) {
             retryStatus.nextAttempt();
+            getStats(callableName, stats -> stats.updateAttempt());
             try {
-                return callable.call();
+                V result = callable.call();
+                getStats(callableName, stats -> stats.updateSuccess());
```

Comment:
I'd add two methods: `recordMetricsOnSuccess()` and `recordMetricsOnFailure()`. It will make the code more compact and descriptive and will be a bit more efficient, because you can lookup the stats once and won't need lambda.